hudi-common/pom.xml

```diff
@@ -97,6 +156,13 @@
   </build>
 
   <dependencies>
+    <!-- Scala -->
+    <dependency>
+      <groupId>org.scala-lang</groupId>
+      <artifactId>scala-library</artifactId>
+      <version>${scala.version}</version>
+    </dependency>
+
```

Comment:
I'm not sure if it's a good idea to have scala deps in hudi-common.  Then there will be hudi-common_2.11 and hudi-common_2.12, which is not really necessary.

Reply:
> I'm not sure if it's a good idea to have scala deps in hudi-common.

Yeah, this is a by-product of keeping this FileIndex in Scala for now. There's also no other module shared by both Spark and Hive unfortunately. 

HUDI-3239 would actually resolve this converting it to Java.

> Then there will be hudi-common_2.11 and hudi-common_2.12, which is not really necessary.

This is not necessary -- since we don't depend on any divergent behavior here

Reply:
Cool